codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -234,17 +478,137 @@ int streamableBytes0(Http2Stream stream) {
         long pseudoTime;
         long totalQueuedWeights;
         private byte flags;
+        short weight = DEFAULT_PRIORITY_WEIGHT;
+
+        State(int streamId) {
+            this(streamId, null, 0);
+        }
 
         State(Http2Stream stream) {
             this(stream, 0);
         }
 
         State(Http2Stream stream, int initialSize) {
+            this(stream.id(), stream, initialSize);
+        }
+
+        State(int streamId, Http2Stream stream, int initialSize) {
             this.stream = stream;
-            queue = new PriorityQueue<State>(initialSize);
+            this.streamId = streamId;
+            pseudoTimeQueue = new DefaultPriorityQueue<State>(StatePseudoTimeComparator.INSTANCE, initialSize);
+        }
+
+        boolean isDescendantOf(State state) {
+            State next = parent;
+            while (next != null) {
+                if (next == state) {
+                    return true;
+                }
+                next = next.parent;
+            }
+            return false;
+        }
+
+        void takeChild(State child, boolean exclusive, List<ParentChangedEvent> events) {
+            takeChild(null, child, exclusive, events);
+        }
+
+        /**
+         * Adds a child to this priority. If exclusive is set, any children of this node are moved to being dependent on
+         * the child.
+         */
+        void takeChild(Iterator<IntObjectMap.PrimitiveEntry<State>> childItr, State child, boolean exclusive,
+                       List<ParentChangedEvent> events) {
+            State oldParent = child.parent;
+
+            if (oldParent != this) {
+                events.add(new ParentChangedEvent(child, oldParent));
+                child.setParent(this);
+                // If the childItr is not null we are iterating over the oldParent.children collection and should
+                // use the iterator to remove from the collection to avoid concurrent modification. Otherwise it is
+                // assumed we are not iterating over this collection and it is safe to call remove directly.
+                if (childItr != null) {
+                    childItr.remove();
+                } else if (oldParent != null) {
+                    oldParent.children.remove(child.streamId);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "oldChild". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)